scripts/combine-profile-information.py

```diff
@@ -0,0 +1,127 @@
+"""
+Script to combine multiple CSVs (each containing profile information of compiling one
+file) into a single summary profile information CSV file.
+"""
+
+from argparse import ArgumentParser
+import csv
+from pathlib import Path
+from typing import Dict, Iterable, Iterator, List, Tuple, Union
+
+parser = ArgumentParser(description="Combine multiple profile CSVs into one")
+parser.add_argument("dump_dir", help="The folder the profile CSVs have been dumped in")
+parser.add_argument("-o", "--summary_path", help="The path to store the summary CSV")
+
+args = parser.parse_args()
+
+DUMP_DIR = Path(args.dump_dir)
+if not DUMP_DIR.exists():
+    raise ValueError(f"{DUMP_DIR} does not exist")
+
+if args.summary_path is None:
+    args.summary_path = DUMP_DIR.parent / "summary.csv"
+SUMMARY_PATH = Path(args.summary_path)
+
+CsvRow = Dict[str, str]
+CsvRows = List[CsvRow]
+SummaryRow = Dict[str, str]
+
+
+def parse_counters(counters: str) -> Dict[str, str]:
+    if not counters:
+        return {}
+    split_counters = counters.strip("[]").split("; ")
+    return dict(map(lambda x: tuple(x.split(" = ")), split_counters))
+
+
+def get_csv_rows(csv_path: Path) -> CsvRows:
+    with open(csv_path) as csv_file:
+        csv_reader = csv.DictReader(csv_file)
+        return list(csv_reader)
+
+
+def get_input_csv_paths() -> Iterator[Tuple[Path, CsvRows]]:
+    return (csv_path for csv_path in DUMP_DIR.glob("*.csv"))
+
+
+def split_into_number_and_unit(number_string: str) -> (Union[int, float], str):
+    numeric = "".join(map(str, range(10))) + "."
+    unit_start_position = next(
+        (i for i, c in enumerate(number_string) if c not in numeric), len(number_string)
+    )
+    number = float(number_string[:unit_start_position])
+    unit = number_string[unit_start_position:]
+    if number.is_integer():
+        number = int(number)
+    return number, unit
+
+
+PRIMARY_KEY = "pass name"
+summary_non_counter_fields = set()
+summary_counter_fields = set()
+
+for rows in map(get_csv_rows, get_input_csv_paths()):
+    curr_fields = set(rows[0])
+    curr_fields.remove(PRIMARY_KEY)
+    if "counters" in curr_fields:
+        curr_fields.remove("counters")
```

Comment:
It feels a bit inconsistent to have a
symbolic constant for the primary
key, but not for counters.